redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -294,83 +294,263 @@ inline void getValues(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
 }
 
 /**
- * Systems derived class for delivering OemPowerSupplyMetrics Schema.
+ * @brief Retrieves valid input history item.
```

Comment:
I found this description a little confusing.  I wasn't sure what the function was doing until reading it a few times.

It seems like it tries to find all the input history object paths for the specified chassis and PSU?  And if one or more are found, it calls the specified callback (vs. returning a value)?

Reply:
Yeah, it attempts to validate that the chassis and power supply being queried actually has a valid input history association.  I modeled it after the getValidPowerSupplyID(...).